### PR TITLE
gaudi: add a patch for missing includes

### PR DESCRIPTION
**Pull request by jmcarcell**
These files have been patched in different PRs with one of them containing other changes so I made a patch with only the necessary changes. Built successfully with GCC 14 on AlmaLinux 9.

### CODE REVIEW

**Comment by wdconinc** [comment]
I assume also here a conflict with gcc@14: for @:36 is necessary?

**Comment by jmcarcell** [comment]
Yes, added.

**Comment by jmcarcell** [comment]
I'm not sure anymore these patches are needed with GCC 14 only since I was hitting one of the issues here when building on AlmaLinux 9 with the system compiler, GCC 11.4, so I have decided to apply the patches for all the compilers.

File: var/spack/repos/builtin/packages/gaudi/package.py [hunk 1]
```diff
@@ -70,6 +70,9 @@ class Gaudi(CMakePackage):
         when="@:38.1",
     )
 
+    # add a few missing includes (c++20?)
+    patch("includes.patch", when="@37:38")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/gaudi/package.py:74]: From your comments it sounds like this is in the master branch so we can close the range.
```suggestion
    patch("includes.patch", when="@37:38.3")
```

**Review by bernhardkaindl — APPROVED** [review]
Confirming that this PR fixes build off gaudi@38 with gcc-11.4.

File: var/spack/repos/builtin/packages/gaudi/package.py (hunk 1 above)
Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/gaudi/package.py:74]: I did not need this incremental change and 38.3 was the last release before 39, so I think this is optional for now.

**Pull request merged**